items/src/main/scala/weco/api/items/services/SierraItemUpdater.scala

```diff
@@ -57,142 +103,123 @@ class SierraItemUpdater(
+      case "harop" => "deepstore"
+      case _       => "library"
+    }
+
     if (accessCondition.exists(_.isRequestable)) {
-      getAvailableDates(accessCondition.head)
-        .map(availableDates => item.copy(availableDates = Some(availableDates)))
+      locationName match {
+        case "deepstore" =>
+          deepstoreItemAvailabilities(getVenuesOpeningTimes(locationName)).map(
+            availableDates =>
+              item.copy(
+                availableDates = Some(availableDates)
+            )
+          )
+        case "library" =>
+          libraryItemAvailabilities(getVenuesOpeningTimes(locationName)).map(
+            availableDates =>
+              item.copy(
+                availableDates = Some(availableDates)
+            )
+          )
+      }
     } else {
       Future.successful(item)
     }
   }
 
-  private def getAvailableDates(
-    accessCondition: DisplayAccessCondition
-  ): Future[List[AvailabilitySlot]] = {
-    val venueFromDisplayAccessCondition = Map(
-      "online-request" -> "library"
-      // other venue to be added as DisplayAccessMethod id -> content-api venue title
-    )
-
-    def getLeadTimeInDays(openingTimes: List[OpenClose]): Int = {
-      val timeAtVenue = LocalDateTime.now(venueClock)
-      val isWorkingDay = timeAtVenue.toLocalDate.isEqual(
-        LocalDate.parse(
-          openingTimes.head.open,
-          DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss.SSSXXX")
-        )
-      )
-      accessCondition.method.id match {
-        case "online-request" if (timeAtVenue.getHour < 10 || !isWorkingDay) =>
-          1
-        case "online-request" if timeAtVenue.getHour >= 10 => 2
-      }
-    }
+  private def getVenuesOpeningTimes(
+    locationName: String
+  ): Future[Map[String, List[AvailabilitySlot]]] =
+    for {
+      venues <- venuesOpeningTimesLookup
+        .byVenueName(locationName)
 
-    venueOpeningTimesLookup
-      .byVenueName(
-        venueFromDisplayAccessCondition.get(accessCondition.method.id)
-      )
-      .map {
-        case Right(venue) =>
-          venue.openingTimes
+      venuesOpeningTimes = venues match {
+        case Right(venues) =>
+          venues
             .map(
-              openClose => AvailabilitySlot(openClose.open, openClose.close)
+              venue =>
+                venue.title -> venue.openingTimes.map(
+                  openingTime =>
+                    AvailabilitySlot(openingTime.open, openingTime.close)
             )
-            // the list of openingTimes, as returned from VenueOpeningTimesLookup, starts at "closest open day"
-            // (including today)
-            // however, it takes ${leadTimeInDays} days for the item to be fetched from stores
-            // so we need to drop ${leadTimeInDays} elements from the list of openingTimes
-            .drop(getLeadTimeInDays(venue.openingTimes))
+          ) toMap
         case Left(venueOpeningTimesLookupError) =>
           error(
```

Comment:
If you move this log into the `venuesOpeningTimesLookup` function you can use the right-biased-ness of Eithers to avoid dealing with the left (you can just map over the Either and ignore dealing with the left as long as you can coerce the Either type to a Map i.e. `.toSeq.toMap`).